crates/wysiwyg/src/composer_model/delete_text.rs

```diff
@@ -87,6 +114,213 @@ where
+                // we are guaranteed to get valid chars here, so can use unwrap
+                let mut current_offset = location.start_offset;
+                let mut current_type = node
+                    .char_type_at_offset(current_offset, &direction)
+                    .unwrap();
+
+                while node.offset_is_inside_node(current_offset, &direction)
+                    && current_type == start_type
+                {
+                    let next_offset = direction.increment(current_offset);
+                    let next_type = node
+                        .char_type_at_offset(current_offset, &direction)
+                        .unwrap();
+
+                    if next_type != current_type {
+                        break;
+                    }
+
+                    current_offset = next_offset;
+                    current_type = next_type;
+                }
+
+                // determine our current position in the dom
+                let current_position = location.position + current_offset;
+
+                if node.offset_is_inside_node(current_offset, &direction) {
+                    // if we have stopped inside the node, first do the required deletion
+                    self.delete_to_cursor(current_position);
+
+                    // if we didn't start this removal at whitespace, stop
+                    if start_type != CharType::Whitespace {
+                        return ComposerUpdate::keep();
+                    }
+
+                    // if we did start with whitespace, get the next set of arguments and make the recursive call
+                    let _args = self.get_remove_word_arguments(&direction);
+                    match _args {
+                        None => ComposerUpdate::keep(),
+                        Some(arguments) => {
+                            let (location, next_type) = arguments;
+                            self.remove_word(
+                                // pass it the new type to remove
+                                next_type, direction, location,
+                            )
+                        }
+                    }
+                } else {
+                    // if we have stopped at the edge of the node, first do the required deletion
+                    self.delete_to_cursor(current_position);
+
+                    // otherwise, make a recursive call to continue to the next node
+                    let _args = self.get_remove_word_arguments(&direction);
+                    match _args {
+                        None => ComposerUpdate::keep(),
+                        Some(details) => {
+                            let (_location, _) = details;
+                            self.remove_word(
+                                // use the original start type from the previous call
+                                start_type, direction, _location,
+                            )
+                        }
+                    }
+                }
```

Comment:
I think that the logic here can be improved due to the common code between the blocks, but I can't figure out the borrowing